File: Leetcode-one-by-one/src/Matrix Similarity After Cyclic Shifts.swift
class Solution {
    func areSimilar(_ mat: [[Int]], _ k: Int) -> Bool {
        let m = mat.count
        let n = mat[0].count
        let k = k % n
        
        for i in 0..<m {
            for j in 0..<n {
                if i % 2 == 0 {
                    if mat[i][j] != mat[i][(j + k) % n] {
                        return false
                    }
                } else {
                    if mat[i][j] != mat[i][(j - k + n) % n] {
                        return false
                    }
                }
            }
        }
        return true
    }
}

File: Leetcode-one-by-one/src/Matrix Similarity After Cyclic Shifts.go
func areSimilar(mat [][]int, k int) bool {
    m := len(mat)
    n := len(mat[0])
    k %= n

    for i := 0; i < m; i++ {
        for j := 0; j < n; j++ {
            if i%2 == 0 {
                if mat[i][j] != mat[i][(j+k)%n] {
                    return false
                }
            } else {
                if mat[i][j] != mat[i][(j-k+n)%n] {
                    return false
                }
            }
        }
    }
    return true
}
